File: packages/papyrus-lang/src/symbols/SymbolBinder.ts
import { Diagnostics } from '../Diagnostics';
import { ScriptNode } from '../parser/Node';
import { SymbolDeclarationVisitor } from './SymbolDeclarationVisitor';

export class SymbolBinder {
    public bindSymbols(scriptNode: ScriptNode, diagnostics: Diagnostics) {
        const definitionVisitor = new SymbolDeclarationVisitor(diagnostics);
        definitionVisitor.visitScript(scriptNode);
    }
}

File: src/DarkId.Papyrus.LanguageService/Program/Symbols/SymbolBinder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DarkId.Papyrus.LanguageService.Syntax;
using DarkId.Papyrus.LanguageService.Syntax.Legacy;

namespace DarkId.Papyrus.LanguageService.Program.Symbols
{
    class SymbolBinder
    {
        public ScriptSymbol Bind(ScriptNode node)
        {
            throw new NotImplementedException();

            //var visitor = new SymbolBindingVisitor(diagnostics);
            //return (ScriptSymbol)visitor.Visit(node, null);
        }
    }
}
